frontend/src/scenes/surveys/surveyLogic.tsx

```diff
@@ -1244,6 +1213,19 @@ export const surveyLogic = kea<surveyLogicType>([
                 return !!enabledFlags[FEATURE_FLAGS.SURVEYS_PARTIAL_RESPONSES]
             },
         ],
+        partialResponsesFilter: [
+            (s) => [s.isPartialResponsesEnabled, s.survey],
+            (isPartialResponsesEnabled: boolean, survey: Survey): string => {
+                if (isPartialResponsesEnabled && survey.enable_partial_responses) {
+                    return `AND uniqueSurveySubmissionsFilter('${survey.id}')`
+                }
+                /**
+                 * Coalescing to True as older events don't have the survey_completed property.
+                 * So they're completed by default since we didn't have partial responses before.
+                 */
+                return `AND COALESCE(JSONExtractBool(properties, '${SurveyEventProperties.SURVEY_COMPLETED}'), True) = True`
+            },
```

Comment:
this is the main query change. If partial responses are enabled (both feature flag and survey setting), then we use the filter as intended. However, if not, we only get responses completed responses